Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Sync/DataSynchronizer.swift

```diff
@@ -1194,16 +1181,19 @@ public class DataSynchronizer: NetworkStateDelegate, FatalErrorListener {
         // contain version information. Clone the document and remove forbidden fields from it before
         // storing it in the collection.
         let docForStorage = DataSynchronizer.sanitizeDocument(remoteDocument)
-        try localCollection.findOneAndReplace(filter: ["_id": documentId], replacement: docForStorage, options: FindOneAndReplaceOptions(upsert: true))
+        try localCollection.findOneAndReplace(filter: ["_id": documentId],
+                                              replacement: docForStorage, options: FindOneAndReplaceOptions(upsert: true))
         try config.setPendingWritesComplete(atVersion: atVersion)
         if documentBeforeUpdate != nil {
             try undoCollection.deleteOne([idField: documentId])
         }
 
-        let event = ChangeEvent<Document>.changeEventForLocalReplace(namespace: namespace,
-                                                                     documentId: documentId,
-                                                                     document: docForStorage,
-                                                                     writePending: false)
+        let event = ChangeEvent<Document>.changeEventForLocalReplace(
+            namespace: namespace,
+            documentId: documentId,
+            document: docForStorage,
+            updateDescription: documentBeforeUpdate?.diff(otherDocument: docForStorage),
```

Comment:
[minor] this goes back to my comment earlier, but a local replace event should not have an updateDescription. A replace is a full document replace, so there is no need for an updateDescription.

Why did this get added?

Reply:
It seems very odd that remote updates that are "coalesced" into replaces lose the update information. This resulted in a weird UX for the app I was building.